### PR TITLE
Scripts/Spells: Slag Behemoth in Blackrock Foundry

### PR DESCRIPTION
**Pull request by ghost**
**Changes proposed:**

-  Add Smart Script for [http://www.wowhead.com/npc=77504/slag-behemoth](Scion Lord Rhyolith in Draenor XD)
-  Add spellscripts for 159750 (Periodic Dummy Aura), 159749 and 159760 dummy effect spells.
-  

**Target branch(es):** 3.3.5/master
- [x] master

**Issues addressed:** Closes #  (insert issue tracker number)

**Tests performed:** (Does it build, tested in-game, etc.)

**Known issues and TODO list:** (add/remove lines as needed)

- [ ] 159757 and 158424 required support for destroy volcanic bomb.
- [ ] If you wipe on this miniboss, bomb also should be destroyed, but SAI don't have support for this.

### CODE REVIEW

File: src/server/scripts/Spells/spell_generic.cpp [hunk 1]
```diff
@@ -4595,6 +4595,74 @@ class spell_gen_impatient_mind : public SpellScriptLoader
...
+
+    class spell_gen_slag_behemoth_blast_waves_AuraScript : public AuraScript
+    {
+        PrepareAuraScript(spell_gen_slag_behemoth_blast_waves_AuraScript);
+
+        void OnPeriodic(AuraEffect const* aurEff)
+        {
+            if (Unit* caster = GetCaster())
+            {
+                if (aurEff->GetTickNumber() > 0 && aurEff->GetTickNumber() < 3)
```
Comment (Keader) [inline src/server/scripts/Spells/spell_generic.cpp]: aurEff->GetTickNumber() can be  < 0 ?

File: src/server/scripts/Spells/spell_generic.cpp [hunk 2]
```diff
@@ -4595,6 +4595,74 @@ class spell_gen_impatient_mind : public SpellScriptLoader
...
+    public:
+        spell_gen_slag_behemoth_blast_waves_dummy_spells() : SpellScriptLoader("spell_gen_slag_behemoth_blast_waves_dummy_spells") { }
+
+        class spell_gen_slag_behemoth_blast_waves_dummy_spells_SpellScript : public SpellScript
+        {
+            PrepareSpellScript(spell_gen_slag_behemoth_blast_waves_dummy_spells_SpellScript);
+
+            void HandleScript(SpellEffIndex /*effIndex*/)
+            {
+                if (Unit* caster = GetCaster())
```
Comment (Keader) [inline src/server/scripts/Spells/spell_generic.cpp]: Caster is never null in SpellScripts

File: src/server/scripts/Spells/spell_generic.cpp [hunk 3]
```diff
@@ -4595,6 +4595,74 @@ class spell_gen_impatient_mind : public SpellScriptLoader
...
+// Called by 159750: TickNumber < 3 should call 159749, TickNumber == 3 should called 159760 and destroy volcanic bomb.
+class spell_gen_slag_behemoth_blast_waves : public SpellScriptLoader
+{
+public:
+    spell_gen_slag_behemoth_blast_waves() : SpellScriptLoader("spell_gen_slag_behemoth_blast_waves") { }
+
+    class spell_gen_slag_behemoth_blast_waves_AuraScript : public AuraScript
+    {
+        PrepareAuraScript(spell_gen_slag_behemoth_blast_waves_AuraScript);
+
```
Comment (Keader) [inline src/server/scripts/Spells/spell_generic.cpp]: Missing validate

File: src/server/scripts/Spells/spell_generic.cpp [hunk 4]
```diff
@@ -4595,6 +4595,74 @@ class spell_gen_impatient_mind : public SpellScriptLoader
...
+            void HandleScript(SpellEffIndex /*effIndex*/)
+            {
+                if (Unit* caster = GetCaster())
+                    caster->CastSpell(caster, uint32(GetEffectValue()));
+            }
+
+            void Register() override
+            {
+                OnEffectHitTarget += SpellEffectFn(spell_gen_subjugator_korul_darkness_calls_SpellScript::HandleScript, EFFECT_0, SPELL_EFFECT_DUMMY);
```
Comment (tkrokli) [inline src/server/scripts/Spells/spell_generic.cpp]: What has Subjugator Kor'ul in Blackfathom Deeps have to do with Slag Behemoth in Blackrock Foundry?

Comment (tkrokli) [inline src/server/scripts/Spells/spell_generic.cpp]: Maybe you based this spell script on the old "darkness calls" spell script, but forgot to change this name?

Comment (Ulduar2) [inline src/server/scripts/Spells/spell_generic.cpp]: Yeah, copypaste mistake from me. Sorry.

File: src/server/scripts/Spells/spell_generic.cpp (hunk 3 above)
Comment (Aokromes) [reply src/server/scripts/Spells/spell_generic.cpp]: @Ulduar2

File: src/server/scripts/Spells/spell_generic.cpp [hunk 5]
```diff
@@ -4595,6 +4595,74 @@ class spell_gen_impatient_mind : public SpellScriptLoader
...
+        PrepareAuraScript(spell_gen_slag_behemoth_blast_waves_AuraScript);
+
+        void OnPeriodic(AuraEffect const* aurEff)
+        {
+            if (Unit* caster = GetCaster())
+            {
+                if (aurEff->GetTickNumber() < 3)
+                    caster->CastSpell(caster, SPELL_BLAST_WAVES_ONE_TICK, TRIGGERED_FULL_MASK, nullptr, aurEff);
+
+                if (aurEff->GetTickNumber() == 3)
```
Comment (Keader) [inline src/server/scripts/Spells/spell_generic.cpp]: else if

File: src/server/scripts/Spells/spell_generic.cpp (hunk 3 above)
Comment (Ulduar2) [reply src/server/scripts/Spells/spell_generic.cpp]: You can close this request. I don't have instrument on current state for check script.
Or you can redone this.

**Pull request closed without merging**